qcp/matrices/sparse_matrix.py

```diff
@@ -471,12 +487,17 @@ def _dot_sparse(self, other: SparseMatrix) -> SparseMatrix:
 
     def __str__(self) -> str:
         total_string = ""
-
         for i in range(self.num_rows):
+            for j in range(self.num_columns):
+                # Remove very small numbers and show neater matrix
+                if float(self[i][j].real+self[i][j].imag) < 10**-15: 
```

Comment:
Could do
```suggestion
                v = self[i][j]
                if isinstance(v, complex) and cmath.isclose(v.real + v.imag, 0):
```
because self[i][j] isn't always guaranteed to be a complex number

Reply:
This is only useful for debugging, I dont think the user will see this. its fine if you can remove this completely

Reply:
Fair point actually